src/software/ai/hl/stp/tactic/goalie/goalie_fsm.cpp

```diff
@@ -222,3 +273,28 @@ void GoalieFSM::moveToGoalLine(const Update &event)
         AutoChipOrKick{AutoChipOrKickMode::OFF, 0.0}, max_allowed_speed_mode, 0.0,
         event.common.robot.robotConstants(), std::optional<double>()));
 }
+
+bool GoalieFSM::retrieveDone(const Update &event)
+{
+    Point ball_position = event.common.world.ball().position();
+    Point retrieve_destination =
+        event.common.world.field().friendlyDefenseArea().centre();
+    return comparePoints(ball_position, retrieve_destination, 0.05);
```

Comment:
Personally I think we can increase this threshold since the exact location doesn't matter. Maybe 0.2meters?!

Reply:
bump @germchow , also make this a class static constant